chall07/mpernia-.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func time_elapse(start time.Time) int {
+	ms := time.Since(start)
+	new_ms := int(ms.Seconds() * 1000)
+	return new_ms
+}
+
+func get_hex(red int, green int, blue int) string {
+	rgb_int := (int64(red) * 256 * 256) + (int64(green) * 256) + int64(blue)
+	return strconv.FormatInt(rgb_int, 16)
+}
+
+func get_link(body string) string {
+	j := strings.IndexByte(body, ',')
+	id, err := strconv.Atoi(body[3:j])
+	if err != nil {
+		print(err)
+	}
+	k := strings.IndexByte(body, 'g')
+	red, err := strconv.Atoi(body[j+3 : k-1])
+	if err != nil {
+		print(err)
+	}
+	j = strings.IndexByte(body, 'b')
+	green, err := strconv.Atoi(body[k+2 : j-1])
+	if err != nil {
+		print(err)
+	}
+	k = strings.IndexByte(body, ' ')
+	blue, err := strconv.Atoi(body[j+2 : k])
+	if err != nil {
+		print(err)
+	}
+	hex := get_hex(red, green, blue)
```

Comment:
There are 2 things you should know here.
- First, it is good to print the errors, but you should exit in someway, make it panic, log a fatal error, it will help prevent unexpected behaviors.
- A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	if _, err := fmt.Sscanf(body, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue); err != nil {
		panic(err)
	}
	return fmt.Sprintf("https://chall07.42madrid.com/?id=%d&resp=%02x%02x%02x", id, red, green, blue)
```